wp-cli/class-theme-check-cli.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+if ( defined( 'WP_CLI' ) && WP_CLI ) {
+	$parent_dir = dirname( __DIR__ );
+	require_once $parent_dir . '/checkbase.php';
+	require_once $parent_dir . '/main.php';
+	WP_CLI::add_command( 'theme-check', 'Theme_Check_CLI' );
+}
+
+class Theme_Check_CLI {
+	/**
+	 * Run a theme check on the specified theme or the current theme.
+	 *
+	 * @param array $args       Indexed array of positional arguments.
+	 * @param array $assoc_args Associative array of options.
+	 * @return void
+	 *
+	 * ## OPTIONS
+	 * [<theme>]
+	 * : The slug of the theme to check. If not provided, checks the current theme.
+	 *
+	 * [--format=<format>]
+	 * : Output format. Accepts 'cli' or 'json'. Default: 'cli'.
+	 *
+	 * ## EXAMPLES
+	 *     wp theme-check run
+	 *     wp theme-check run twentytwentyfour
+	 *     wp theme-check run --format=json
+	 *     wp theme-check run twentytwentyfour --format=json
+	 */
+	public function run( $args, $assoc_args ) {
+		// Get the output format
+		$format = \WP_CLI\Utils\get_flag_value( $assoc_args, 'format', 'cli' );
+		if ( ! in_array( $format, array( 'cli', 'json' ) ) ) {
+			WP_CLI::error( "Invalid format. Accepts 'cli' or 'json'." );
+			return;
+		}
+
+		// Get the current theme
+		$current_theme      = wp_get_theme();
+		$current_theme_slug = $current_theme->get_stylesheet();
+
+		// Use the provided theme slug if available, otherwise use the current theme
+		$check_theme_slug = ! empty( $args[0] ) ? $args[0] : $current_theme_slug;
+
+		// Get the theme
+		$theme = wp_get_theme( $check_theme_slug );
+
+		if ( ! $theme->exists() ) {
+			if ( $format === 'json' ) {
+				$json_output = array(
+					'check-completed' => false,
+					'result'          => "Error: Theme '{$check_theme_slug}' not found.",
+					'messages'        => array(),
+				);
+				WP_CLI::log( wp_json_encode( $json_output, JSON_PRETTY_PRINT ) );
+				return;
+			}
+			WP_CLI::error( "Theme '{$check_theme_slug}' not found." );
+			return;
```

Comment:
WP_CLI::error() exits the process. So I don't think `return` is needed.

Reply:
:+1: removed here: https://github.com/WordPress/theme-check/pull/458/commits/37542b38dd2ac7b8d2125f67dcc4cfc59dfa4541